questionpy_server/cache.py

```diff
@@ -0,0 +1,152 @@
+from collections import OrderedDict
+from pathlib import Path
+from typing import NamedTuple
+
+
+class File(NamedTuple):
+    path: Path
+    size: int
+
+
+class SizeError(Exception):
+    def __init__(self, message: str = '', max_size: float = 0, actual_size: float = 0):
+        super().__init__(message)
+
+        self.max_size = max_size
+        self.actual_size = actual_size
+
+
+class FileLimitLRU:
+    """
+    Limit file cache size, evicting the least recently accessed file when the specified maximum is exceeded.
+    Only `bytes` type values are accepted. Their size is calculated by passing them into the builtin `len()` function.
+    """
+
+    def __init__(self, directory: str, max_bytes: int, extension: str = None) -> None:
+        self.max_bytes = max_bytes
+        self._total_bytes = 0
+        self._extension: str = '' if extension is None else '.' + extension.lstrip('.')
+
+        self._files: OrderedDict[str, File] = OrderedDict()
+
+        self._directory: Path = Path(directory).resolve()
+        self._directory.mkdir(exist_ok=True)
```

Comment:
Hier vielleicht auch die Elternverzeichnisse mit erstellen?

```suggestion
        self._directory.mkdir(exist_ok=True, parents=True)
```

Reply:
Gerne können wir hier noch die Elternverzeichnisse erstellen - ich persönlich finde es beruhigender, wenn ein Fehler ausgegeben wird, falls diese nicht existieren.
So kann ich sicher sein, den richtigen Pfad in den Settings angegeben zu haben (vor allem bei relativen Pfaden).

Reply:
Alles klar :+1: